framework/wazuh/core/batcher/mux_demux.py

```diff
@@ -0,0 +1,249 @@
+        return message
+
+    def send_to_demux(self, msg: Message):
+        """Puts a message into the demux queue.
+
+        Parameters
+        ----------
+        msg : Message
+            The message to be put into the demux queue.
+        """
+        self.demux_queue.put(msg)
+
+    def is_response_pending(self, uid: int) -> bool:
+        """Checks if a response is available for a given unique identifier.
+
+        Parameters
+        ----------
+        uid : int
+            The unique identifier to check.
+
+        Returns
+        -------
+        bool
+            True if the response is available, False otherwise.
+        """
+        return uid not in self.responses
+
+    def receive_from_demux(self, uid: int) -> Optional[dict]:
+        """Retrieves and removes a response from the dictionary for a given unique identifier.
+
+        Parameters
+        ----------
+        uid : int
+            The unique identifier for the response.
+
+        Returns
+        -------
+        Optional[dict]
+            The response dictionary if available, None otherwise.
+        """
+        if not self.is_response_pending(uid):
+            response = self.responses[uid]
+            del self.responses[uid]
+            return response
+        else:
+            return None
+
+    def internal_response_from_demux(self) -> Message:
+        """Retrieves a message from the demux queue.
+
+        Returns
+        -------
+        Message
+            The message retrieved from the demux queue.
+        """
+        return self.demux_queue.get()
+
+    def internal_store_response(self, msg: Message):
+        """Updates the responses dictionary with the message content.
+
+        Parameters
+        ----------
+        msg : Message
+            The message whose content will be added to the responses dictionary.
+        """
+        self.responses[msg.uid] = msg.msg
+
+
+class MuxDemuxRunner(Process):
+    """A multiprocessing Process that manages the MuxDemuxQueue operations, handling
+    signals for graceful shutdown and processing items from the queue.
+
+    The MuxDemuxRunner class runs as a separate process to manage the `MuxDemuxQueue`. It listens
+    for termination signals (SIGTERM, SIGINT) to initiate a graceful shutdown, and continuously
+    processes items from the queue, storing them as responses if they are of the correct type.
+    """
+
+    def __init__(self, queue: MuxDemuxQueue):
+        super().__init__()
+        self.queue = queue
+        self._shutdown_event = Event()
+
+    def _handle_signal(self, signum: int, frame: Any):
+        """Handles termination signals (SIGTERM, SIGINT) by setting the shutdown event.
+
+        Parameters
+        ----------
+        signum : int
+            The signal number received.
+        frame : Any
+            The current stack frame (unused).
+        """
+        logger.info(f'MuxDemuxQueue pid {os.getpid()} - received signal {signum}')
```

Comment:
Same here

```suggestion
        logger.info(f'MuxDemuxQueue pid {os.getpid()} - received signal {signal.Signals(signum).name}')
```

Reply:
Fixed in cc8e51ccec60b8e9576fb54270b2182bef34530f.